src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -291,6 +291,41 @@ public int CompareTo(object? value)
 
         public static TimeSpan FromDays(double value) => Interval(value, TicksPerDay);
 
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> that represents a specified number of days.
+        /// </summary>
+        /// <param name="days">A number of days.</param>
+        /// <returns>An object that represents <paramref name="days"/>.</returns>
+        /// <exception cref="OverflowException">
+        /// <paramref name="days"/> is less than <see cref="MinValue"/> or greater than <see cref="MaxValue"/>.
+        /// </exception>
+        public static TimeSpan FromDays(int days)
+        {
+            return FromMicroseconds(days * MicrosecondsPerDay);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> that represents a specified number of days.
+        /// </summary>
+        /// <param name="days">A number of days.</param>
+        /// <param name="hours">A number of hours.</param>
+        /// <param name="minutes">A number of minutes.</param>
+        /// <param name="seconds">A number of seconds</param>
+        /// <param name="milliseconds">A number of milliseconds</param>
+        /// <param name="microseconds">A number of microseconds</param>
+        /// <returns>An object that represents <paramref name="days"/>, <paramref name="hours"/>, <paramref name="minutes"/>, <paramref name="seconds"/>, <paramref name="milliseconds"/>, <paramref name="microseconds"/>.</returns>
+        /// <exception cref="OverflowException">
+        /// </exception>
+        public static TimeSpan FromDays(int days, int hours = 0, long minutes = 0, long seconds = 0, long milliseconds = 0, long microseconds = 0)
+        {
+            return FromDays(days)
+                + FromHours(hours)
+                + FromMinutes(minutes)
+                + FromSeconds(seconds)
+                + FromMilliseconds(milliseconds)
+                + FromMicroseconds(microseconds);
+        }
```

Comment:
What happens if e.g. someone passes a days value that would overflow but then a negative hours value that would bring it back into the valid range? With this implementation that will still throw... is that the desired behavior?

Reply:
This is an excellent question. We may implement this inside `checked` block and do the math manually by converting all units to ticks and adding them. I know we'll sacrifice some perf, but we'll have the correct behavior.

Reply:
Personally I think that we should only throw if the final tick count is out of range.

It's always annoying when `x + y + z` fails but `x + z + y` would succeed and I think it provides a better experience if we treat `FromDays` as its own higher order thing.